src/codegen/sys/ffi_type.rs

```diff
@@ -161,11 +160,11 @@ fn fix_name(env: &Env, type_id: library::TypeId, name: &str) -> Result {
         match *env.library.type_(type_id) {
             Type::Array(..) | Type::PtrArray(..)
                     | Type::List(..) | Type::SList(..) | Type::HashTable(..) => {
-                if Some(MAIN_NAMESPACE) == env.library.find_namespace("GLib") {
+                if env.namespaces.glib_is_main {
                     Ok(name.into())
                 }
                 else {
-                    Ok(format!("{}::{}", crate_name("GLib"), name))
+                    Ok(format!("{}::{}", &env.namespaces[type_id.ns_id].crate_name, name))
```

Comment:
If it really right change ( `env.library.find_namespace("Glib") == type_id.ns_is` ) then not better use same as in lines 173-177 (`let name_with_prefix = ...`) ?